browser/modules/SavantShieldStudy.jsm

```diff
@@ -176,4 +185,65 @@ class TelemetryEvents {
   }
 }
 
+class BookmarkObserver {
+  constructor(studyCategory) {
+    this.STUDY_TELEMETRY_CATEGORY = studyCategory;
+    // there are two probes: bookmark and follow_bookmark
+    this.METHOD_1 = "bookmark";
+    this.EXTRA_SUBCATEGORY_1 = "feature";
+    this.METHOD_2 = "follow_bookmark";
+    this.EXTRA_SUBCATEGORY_2 = "navigation";
+    // nsINavBookmarksService constant
+    this.TYPE_BOOKMARK = 1;
+  }
+
+  init() {
+    this.observer = {
+      onItemAdded: (itemID, parentID, index, itemType, uri, title, dateAdded, guid, parentGUID) => {
+        if (itemType === this.TYPE_BOOKMARK) {
```

Comment:
You probably want to also check && !uri.schemeIs("place") to exclude special places queries.
Also in onItemRemoved

Reply:
Also, note that most of these notifications have a source argument you can use to exclude bookmarks modified by: sync, import, restore.
Pretty much you could just check source == PlacesUtils.bookmarks.SOURCE_DEFAULT

Reply:
Thanks for the comments; for the first one, what is a bookmark with a `"place"` scheme for? When would a `place://...` URI bookmark be added? Just trying to understand what the check is about.

That is a great suggestion for the second point; I have added that check. Looks like the docs in MDN are not up to date? I don't see a "source" argument there for [onItemAdded](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsINavBookmarkObserver#onItemAdded()) or [onItemRemoved](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsINavBookmarkObserver#onItemRemoved()).

Reply:
MDN is terribly out-of-date, the .idl inline documentation is usually much better :(

"place:query" uris (actually without //) are special uris for internal use that allow to create containers like Most Visited or Recently Bookmarked. We add a couple of those as default bookmarks, they should be pretty rare and it's unlikely for users to create them manually.
Filtering them out doesn't hurt, but even not doing that is unlikely to skew your stats since they are rare.